### PR TITLE
Add new benchmarks

### PR DESCRIPTION
**Pull request by louiscryan**
@ejona86 

Add additional benchmarks including:

- streaming ping-pong messages per second
- unary response bandwidth in megabits per second
- streaming response bandwidth in megabits per second

### CODE REVIEW

**Comment by louiscryan** [comment]
@nmittler - FYI as you're refactoring  benchmarks

File: benchmarks/src/jmh/java/io/grpc/benchmarks/netty/FlowControlledMessagesPerSecondBenchmark.java [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+    callCounter = new AtomicLong();
+    completed = new AtomicBoolean();
+    startFlowControlledStreamingCalls(maxConcurrentStreams, callCounter, completed, 1);
+  }
+
+  /**
+   * Stop the running calls then stop the server and client channels.
+   */
+  @TearDown(Level.Trial)
+  public void stopChannelsAndServers() throws Exception {
```
Comment (nmittler) [inline benchmarks/src/jmh/java/io/grpc/benchmarks/netty/FlowControlledMessagesPerSecondBenchmark.java]: nit: maybe just `teardown()`?

Comment (nmittler) [inline benchmarks/src/jmh/java/io/grpc/benchmarks/netty/FlowControlledMessagesPerSecondBenchmark.java]: Oh I see ... I think you're missing `@Override`. Maybe rename the base method too?

File: benchmarks/src/jmh/java/io/grpc/benchmarks/netty/StreamingPingPongsPerSecondBenchmark.java [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+  }
+
+  /**
+   * Measure throughput of unary calls. The calls are already running, we just observe a counter
+   * of received responses.
+   */
+  @Benchmark
+  public void pingPong(AdditionalCounters counters) throws Exception {
+    // No need to do anything, just sleep here.
+    Thread.sleep(1001);
```
Comment (nmittler) [inline benchmarks/src/jmh/java/io/grpc/benchmarks/netty/StreamingPingPongsPerSecondBenchmark.java]: Not sure I follow ... where is the request being sent?

File: benchmarks/src/jmh/java/io/grpc/benchmarks/netty/StreamingResponseBandwidthBenchmark.java [hunk 3]
```diff
@@ -0,0 +1,103 @@
...
+    super.stopChannelsAndServers();
+  }
+
+  /**
+   * Measure bandwidth of streamed responses.
+   */
+  @Benchmark
+  public void stream(AdditionalCounters counters) throws Exception {
+    // No need to do anything, just sleep here.
+    Thread.sleep(1001);
```
Comment (nmittler) [inline benchmarks/src/jmh/java/io/grpc/benchmarks/netty/StreamingResponseBandwidthBenchmark.java]: Same question .... where is the rpc?

File: benchmarks/src/jmh/java/io/grpc/benchmarks/netty/StreamingPingPongsPerSecondBenchmark.java (hunk 2 above)
Comment (nmittler) [inline benchmarks/src/jmh/java/io/grpc/benchmarks/netty/StreamingPingPongsPerSecondBenchmark.java]: Oh ok, so `startStreamingCalls` gets everything going, so we're not measuring throughput. We're just stopping after 1 second and checking the results.  Is my understanding correct?

File: benchmarks/src/jmh/java/io/grpc/benchmarks/netty/StreamingResponseBandwidthBenchmark.java (hunk 3 above)
Comment (nmittler) [inline benchmarks/src/jmh/java/io/grpc/benchmarks/netty/StreamingResponseBandwidthBenchmark.java]: nm ... see previous reply.

**Comment by nmittler** [comment]
@louiscryan LGTM

File: benchmarks/src/jmh/java/io/grpc/benchmarks/netty/FlowControlledMessagesPerSecondBenchmark.java (hunk 1 above)
Comment (louiscryan) [inline benchmarks/src/jmh/java/io/grpc/benchmarks/netty/FlowControlledMessagesPerSecondBenchmark.java]: Done, here and elsewhere

File: benchmarks/src/jmh/java/io/grpc/benchmarks/netty/StreamingPingPongsPerSecondBenchmark.java (hunk 2 above)
Comment (louiscryan) [inline benchmarks/src/jmh/java/io/grpc/benchmarks/netty/StreamingPingPongsPerSecondBenchmark.java]: Yup, we're measuring throughput, just not in this thread.

**Pull request merged**